### PR TITLE
set AMBIENT_TYPE env var for ztunnel

### PR DESCRIPTION
**Pull request by GregHanson**
**Please provide a description of this PR:**

related ztunnel PR:
https://github.com/istio/ztunnel/pull/296

sets env var for ztunnel that istiod checks for in internal `IsZTunnel()` logic

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ X ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Comment by GregHanson** [comment]
related failure: https://github.com/istio/istio/pull/42506

**Review by stevenctl — APPROVED** [review]

File: manifests/charts/ztunnel/templates/daemonset.yaml [hunk 1]
```diff
@@ -250,6 +250,8 @@ spec:
         - proxy
         - sidecar
         env:
+        - name: AMBIENT_TYPE
```
Comment (stevenctl) [inline manifests/charts/ztunnel/templates/daemonset.yaml:253]: Should this be ISTIO_META_AMBIENT_TYPE?

Comment (GregHanson) [reply manifests/charts/ztunnel/templates/daemonset.yaml:253]: istio agent has logic for grabbing all env var prefixed with `ISTIO_META_` prefix ([here](https://github.com/istio/istio/blob/master/pkg/bootstrap/config.go#L534)).  ztunnel currently doesn't have that logic (https://github.com/istio/ztunnel/pull/296) - was not sure if that was a function we want added in ztunnel

Comment (GregHanson) [reply manifests/charts/ztunnel/templates/daemonset.yaml:253]: @howardjohn thoughts on whether we need similar env var metadata collection in ztunnel?

Comment (howardjohn) [reply manifests/charts/ztunnel/templates/daemonset.yaml:253]: Might be useful eventually but there are currently no cases when we need to pass metadata

**Comment by GregHanson** [comment]
resolved pending discussion, removing `no-not-merge`

**Pull request merged**